airflow/models/baseoperator.py

```diff
@@ -382,7 +389,16 @@ def __init__(
                 stacklevel=3
             )
         validate_key(task_id)
-        self.task_id = task_id
+        self.label = task_id
+
+        # Prefix task_id with group_id
+        task_group = task_group or TaskGroupContext.get_current_task_group(dag)
+        if task_group:
+            self.task_id = f"{task_group.group_id}.{self.label}" if task_group.group_id else self.label
```

Comment:
it looks like we are still changing the semantic of task_id here. would it make more sense to use `self.label` to store ` f"{task_group.group_id}.{self.label}"` and leave `task_id` as is?

Reply:
Hi @houqp  `self.task_id` is only prefixed with `group_id` if the DAG starts to use TaskGroup. For existing dags not using TaskGroup, `self.task_id` remains equal to the passed in `task_id` and are unchanged.
 
The reason I'm prefixing `self.task_id` with `group_id` is to avoid duplicated task_id in `dag.task_dict`. E.g, if we write code like this:
```

def create_section():
    task1 = DummyOperator(task_id="task1")
    task2 = DummyOperator(task_id="task2")

with DAG(...) as dag:
    with TaskGroup("section1") as section1:
        create_section()

    with TaskGroup("section2") as section2:
        create_section()
```

The actual task_id stored in dag.task_dict will be this. This is to avoid duplication.
```
section1.task1
section1.task2
section2.task1
section2.task2
```
The label users need to see in Graph View is this. I.e. they don't need to see the fully-qualified task_id 
in the graph because the task is nested within the group.

```
section1
    task1
    task2
section2
    task1
    task2
```


Alternatively, if we want to avoid changing the semantic of `self.task_id` here when using TaskGroup. We can ask the user to pass in `label` separately and let the user modify the task_id themselves to avoid duplication.
I find this introduces extra burden to the user. However, maybe there are advantages to this too? What do you think? 

```
def create_section(group_id):
    task1 = DummyOperator(task_id=f"{group_id}.task1", label="task1")
    task2 = DummyOperator(task_id="{group_id}.task2", label="task2")

with DAG(...) as dag:
    with TaskGroup("section1") as section1:
        create_section(section1.group_id)

    with TaskGroup("section2") as section2:
        create_section(section2.group_id)
```

Reply:
I see where you are coming from now. I get that the semantic of task_id will remain the same if task group is not being used. I am trying to see if we can maintain that for task grouped tasks as well so it will be less surprise for people when they adopt it.

I agree with you that asking users to manually label tasks is not a good experience and we should not go that route.

It looks like the core of the problem is task names could conflict between groups. Is that an option to simply not allow users define tasks with conflicting names across groups? Or do we have to support task_id namespacing by group due to how it's being handled in subdag operator? I feel like this will simplify the design a lot and make it truly a UI-only feature.

Reply:
Hi @houqp, even without TaskGroup, users cannot define tasks with conflicting names in the same DAG. So if we simply remove the notion of `label` in this PR, we can achieve that. That means the label in Graph View will be the same as `task_id`. And users have to do their own work to maintain unique `task_id` across the entire DAG, and across TaskGroups.

The same example will look like this. And the `task_id` kept in `dag.task_dict` and on the Graph View are like this:
```
section_1_task_1
section_1_task_2
section_2_task_1
section_2_task_2
```

```
def create_section(group_id):
    task1 = DummyOperator(task_id=f"{group_id}_task_1")
    task2 = DummyOperator(task_id=f"{group_id}_task_2")

with DAG(...) as dag:
    with TaskGroup("section1") as section1:
        create_section(section1.group_id)

    with TaskGroup("section2") as section2:
        create_section(section2.group_id)
```

How does it sound?

Reply:
> So if we simply remove the notion of label in this PR, we can achieve that. That means the label in Graph View will be the same as task_id. And users have to do their own work to maintain unique task_id across the entire DAG, and across TaskGroups.

This is what I have in mind as well since we already require users to make sure task_ids are unique today. I previous comment as asking whether there is some sort of special requirement in subdag support that requires us to support task namespacing within the group. Based on your answer, it looks like there isn't.

I would prefer we go with the simpler route, i.e. dropping label, if it meets all our existing requirements. So far, it maintains task_id semantics and comes with lower code complexity, so it's a win win to me :)

Reply:
Actually I think you have a really good point. In fact, after trying this out over some real existing DAGs, I realized leaving `task_id` fully to the user is better. For existing DAGs, when people want to convert to use TaskGroups, they don't want to modify the `task_id` (because doing so causes the tasks to appear as new tasks in the DAG).

So I took your suggestion:
- Removed `label`
- Do not prefix `task_id` with group_id
- Require `group_id` and `task_id` to be unique throughout the whole DAG

Reply:
> Do not prefix task_id with group_id

This is a downgrade from current SubDagOperator. You can have duplicated task_id across subdags in the same dag at the moment. I would suggest actually prefixing.

Also this change is a divergence with what's voted and proposed in https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-34+TaskGroup%3A+A+UI+task+grouping+concept+as+an+alternative+to+SubDagOperator. I would either bring this up for discussion on the mailing list and wait for a lazy consensus, or refactor AIP and re-submit for vote.

Reply:
Hi @casassg . Thanks for the suggestion. I brought this up on the email "[AIP-34] Rewrite SubDagOperator". Let's see what most people say.